vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -731,8 +731,24 @@ import { ItemCache } from '@vaadin/vaadin-grid/src/vaadin-grid-data-provider-mix
           }
           updateGridItemsInDomBasedOnCache(items);
         }
+        const cacheAndIndex = grid._cache.getCacheAndIndexByKey(pkey);
+        clearSubcache(cacheAndIndex.cache, cacheAndIndex.scaledIndex, index, length);
       });
 
+      /*
+       * Deletes items from grid cache, including the descendants of the affected
+       * items.
+       */
+      function clearSubcache(cache,scaledIndex, index, length) {
+        const end = length + index;
+        const childCache = cache.itemCaches[scaledIndex];
+        for(let itemIndex = index; itemIndex < end; itemIndex++) {
+          delete childCache.items[itemIndex];
+          delete childCache.itemCaches[itemIndex];
+        }
+        cache.updateSize();
```

Comment:
This will update the parent cache's effective size, but that's only one level upwards from the child cache. It works in case your grid only has 2 levels but with any deeper hierarchy, the root level effective size would not get updated.

If you want to keep the cache size update here in the same function, you'll also need to pass the root level cache as a parameter.

Reply:
Could be cleaner to just update it at the very end of `clear(index, length, parentKey)` (but would it make this function incomplete?)

Reply:
Removed the function.